### PR TITLE
[BUG] Incomplete BingX "trade-new-order.mjs" implementation. Solution only requires a few more lines of code. #6351

### PR DESCRIPTION
**Pull request by lcaresia**
### <samp>🤖 Generated by Copilot at 5f756de</samp>

Added optional parameters for take profit and stop loss prices to two BingX actions, `custom-bracket-order` and `trade-new-order`, to enable more advanced trading strategies. Updated the BingX package version, the `pnpm-lock.yaml` file, and added three new components to the Pipedream platform.

### <samp>🤖 Generated by Copilot at 5f756de</samp>

> _BingX package grows_
> _New parameters for trades_
> _`pnpm-lock` changes_

### <samp>🤖 Generated by Copilot at 5f756de</samp>

*  Bump up the version of the BingX package and update the `lodash` dependency in the `pnpm-lock.yaml` file ([link](https://github.com/PipedreamHQ/pipedream/pull/6367/files?diff=unified&w=0#diff-37e6e279bc7d94ca0ed853df4c0467bb6a7c0cfcb8f17e156323f8f896cd1ae5L3-R3), [link](https://github.com/PipedreamHQ/pipedream/pull/6367/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL1147-R1153))
*  Add two new optional parameters, `takerProfitPrice` and `stopLossPrice`, to the BingX Custom Bracket Order and Trade New Order actions, and pass them to the API request parameters ([link](https://github.com/PipedreamHQ/pipedream/pull/6367/files?diff=unified&w=0#diff-1cd1ab9169c504e0793106a152db4c719fdd4acde0f6e616adf0d3aae7cc3480R65-R76), [link](https://github.com/PipedreamHQ/pipedream/pull/6367/files?diff=unified&w=0#diff-1cd1ab9169c504e0793106a152db4c719fdd4acde0f6e616adf0d3aae7cc3480R109-R110), [link](https://github.com/PipedreamHQ/pipedream/pull/6367/files?diff=unified&w=0#diff-1cd1ab9169c504e0793106a152db4c719fdd4acde0f6e616adf0d3aae7cc3480L107-R124), [link](https://github.com/PipedreamHQ/pipedream/pull/6367/files?diff=unified&w=0#diff-e3ce12deae82dd360db064c144edcb44b0459f2bc6b57de7e0f4cfa5b35a0937L46-R57), [link](https://github.com/PipedreamHQ/pipedream/pull/6367/files?diff=unified&w=0#diff-e3ce12deae82dd360db064c144edcb44b0459f2bc6b57de7e0f4cfa5b35a0937R70-R71))
*  Bump up the version of the BingX Custom Bracket Order and Trade New Order actions to reflect the new features ([link](https://github.com/PipedreamHQ/pipedream/pull/6367/files?diff=unified&w=0#diff-1cd1ab9169c504e0793106a152db4c719fdd4acde0f6e616adf0d3aae7cc3480L5-R5), [link](https://github.com/PipedreamHQ/pipedream/pull/6367/files?diff=unified&w=0#diff-e3ce12deae82dd360db064c144edcb44b0459f2bc6b57de7e0f4cfa5b35a0937L5-R5))
*  Add new components, `h_supertools_analytics_tool`, `implisense_api`, and `liondesk`, to the `pnpm-lock.yaml` file ([link](https://github.com/PipedreamHQ/pipedream/pull/6367/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1800-R1802), [link](https://github.com/PipedreamHQ/pipedream/pull/6367/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1961-R1963), [link](https://github.com/PipedreamHQ/pipedream/pull/6367/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR2237-R2239))

### CODE REVIEW

File: components/bingx/actions/custom-bracket-order/custom-bracket-order.mjs [hunk 1]
```diff
@@ -2,7 +2,7 @@ import bingx from "../../bingx.app.mjs";
 
 export default {
   name: "BingX Custom Bracket Order",
-  version: "0.0.3",
+  version: "0.0.4",
```
Comment (GTFalcao) [inline components/bingx/actions/custom-bracket-order/custom-bracket-order.mjs:5]: A new feature/addition should warrant a new `minor` version [as per the guidelines](https://pipedream.com/docs/components/guidelines/#versioning), the same goes for the other component as well as the package version.
```suggestion
  version: "0.1.0",
```

File: components/bingx/actions/trade-new-order/trade-new-order.mjs [hunk 2]
```diff
@@ -2,7 +2,7 @@ import bingx from "../../bingx.app.mjs";
 
 export default {
   name: "BingX Trade New Order",
-  version: "0.0.3",
+  version: "0.0.4",
```
Comment (GTFalcao) [inline components/bingx/actions/trade-new-order/trade-new-order.mjs:5]:
```suggestion
  version: "0.1.0",
```

File: components/bingx/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@pipedream/bingx",
-  "version": "0.0.7",
+  "version": "0.0.8",
```
Comment (GTFalcao) [inline components/bingx/package.json:3]:
```suggestion
  "version": "0.1.0",
```

**Review by GTFalcao — COMMENTED** [review]
Hi @lcaresia , thanks for the contribution!

I've commented on version numbers that need to be adjusted, following the semantic versioning guidelines. Since this is unrelated to the actual code and functionality, I'm moving the issue forward.

**Review by GTFalcao — APPROVED** [review]

**Comment by zaiden96** [comment]
Hi @lcaresia and @GTFalcao - please see the following error:

image


I think deleting the ".replace(",", ".")" from lines 65 and 69 may be enough to resolve the bug since the format of the "string" will be the same for take profit/stop loss as it is for entrustPrice and entrustVolume, so the replace function may be unnecessary:

image

**Comment by lcaresia** [comment]
@zaiden96 Are you yet experiencing this error? if necessary I open a new task to solve this! Thanks!

**Comment by zaiden96** [comment]
@lcaresia another task was opened here and a proposed solution is pending QA: https://github.com/PipedreamHQ/pipedream/issues/6503#issuecomment-1550532364

**Comment by lcaresia** [comment]
@zaiden96 Perfect!!

**Pull request merged**